addons/mail/controllers/discuss/thread.py

```diff
@@ -1,20 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo.addons.mail.controllers.thread import ThreadController
-
-
-class DiscussThreadController(ThreadController):
-    def _filter_message_post_partners(self, thread, partners):
-        if thread._name != "discuss.channel":
-            return super()._filter_message_post_partners(thread, partners)
-        if thread.channel_type == "channel":
-            if group := ((thread.parent_channel_id or thread).group_public_id):
-                domain = [("id", "in", partners.ids), ("user_ids.all_group_ids", "in", group.ids)]
-                # sudo: res.partner - filtering partners that have the correct group is acceptable.
-                partners = self.env["res.partner"].sudo().search(domain)
```

Comment:
Note: this part is moved as `message_post` override as it should apply in all flows, it's not related to controller params validation.